presto-main/src/main/java/io/prestosql/sql/planner/LocalDynamicFilter.java

```diff
@@ -52,20 +53,26 @@
     // Mapping from dynamic filter ID to its build channel indices.
     private final Map<String, Integer> buildChannels;
 
-    private final SettableFuture<TupleDomain<Symbol>> resultFuture;
+    // Mapping from dynamic filter ID to its build channel type.
```

Comment:
I don't think you need to modify `LocalDynamicFilter` (`TupleDomain` works here fine)

I think you only need to fix `LocalDynamicFiltersCollector`